src/main/java/bio/terra/service/dataset/AzureStorageResource.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.service.dataset;
+
+import bio.terra.app.model.AzureCloudResource;
+import bio.terra.app.model.AzureRegion;
+import bio.terra.app.model.CloudRegion;
+import bio.terra.app.model.CloudResource;
+import bio.terra.model.CloudPlatform;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.Objects;
+import java.util.UUID;
+
+@JsonTypeName("azure")
+public class AzureStorageResource implements StorageResource {
+
+    private UUID datasetId;
+    private CloudPlatform cloudPlatform;
+    private AzureCloudResource cloudResource;
+    private AzureRegion region;
+
+    AzureStorageResource() {
+        this.cloudPlatform = CloudPlatform.AZURE;
+    }
+
+    @Override
+    public UUID getDatasetId() {
+        return datasetId;
+    }
+
+    @Override
+    public AzureStorageResource datasetId(UUID datasetId) {
+        this.datasetId = datasetId;
+        return this;
+    }
+
+    @Override
+    public CloudPlatform getCloudPlatform() {
+        return cloudPlatform;
+    }
+
+    @Override
+    public CloudResource getCloudResource() {
```

Comment:
Can return a subtype here, due to java's covariant return type support.
```suggestion
    public AzureCloudResource getCloudResource() {
```

Reply:
I don't so. I tried originally and it got really mad at me.  I suppose I could by adding generics to the interface...that would avoid having to use casts, which would be nice

Reply:
What errors did you get? I checked out your branch and added this change and it seems to work, although I didn't run all the tests.

Reply:
Ah, it breaks in the setter

Reply:
Ah, fun with generics fixes this!  I'll push up some changes in a few minutes

Reply:
You shouldn't need to use generics to fix this. Like I said, I checked out your code, made this change, and then compiled and started to run the tests. I could run all the tests if you think it will fail later.

Reply:
> Ah, it breaks in the setter

Oh, so if the types don't match then the JSON serialization breaks? Oh well.